lib/ansible/module_utils/urls.py

```diff
@@ -602,7 +602,7 @@ def get_ca_certs(self):
         to_add = False
 
         # Write the dummy ca cert if we are running on Mac OS X
-        if system == 'Darwin':
+        if system == b('Darwin'):
```

Comment:
This is on python 3.6, OSX 10.12. I can consistently reproduce it by adding/removing these patches. It works fine in my other OS, Ubuntu. I don't have access to other python versions on OSX, unfortunately.